File: src/ackward/datetime/Module.cpp
#include <ackward/datetime/Module.hpp>

#include <boost/python/extract.hpp>
#include <boost/python/import.hpp>

using namespace boost::python;

namespace ackward { namespace datetime
{

object module()
{
    static object mod;
    static bool initialized = false;

    if (!initialized)
    {
        mod = import("datetime");
        initialized = true;
    }

    return mod;
}

unsigned int MINYEAR()
{
    return extract<unsigned int>(
        module().attr("MINYEAR"));
}

unsigned int MAXYEAR()
{
    return extract<unsigned int>(
        module().attr("MAXYEAR"));
}

}}

File: src/ackward/re/Module.py
from ackward import (Module,
                     Namespace,
                     TranslationUnit)

def tunit():
    return TranslationUnit(
        guard='INCLUDE_ACKWARD_RE_MODULE_HPP')

def definition():
    with tunit() as t:
        with Namespace('ackward', 're'):
            Module(name='re')

        # objs = [
        #     Module(name='re'),
        #     # function('RegexObject compile(boost::python::str pattern, unsigned char flags=0x00)'),
        #     # function('MatchObject search(boost::python::str pattern, boost::python::str string, unsigned char flags=0x00)'),
        #     # function('MatchObject match(boost::python::str pattern, boost::python::str string, unsigned char flags=0x00)'),
        #     # function('std::vector<boost::python::str> split(boost::python::str pattern, boost::python::str string, short maxsplit=0')
        #     ]

        # super(REModule, self).__init__(
        #     header_includes=[
        #         ('boost', 'python', 'str.hpp'),
        #         # ('ackward', 're', 'RegexObject.hpp')
        #         ]
        #     objects={ ('ackward', 're') : objs })

    return t
